modules/business-text/business-text.service.js

```diff
@@ -37,32 +39,122 @@ class BusinessTextService {
       };
     }
 
-    const text = await BusinessText.findByIdAndUpdate(id, businessText, {
+    const newPage = files.length
+      ? await this.pageImageReplaceHandler(businessText, files)
+      : businessText;
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const newImages = this.imageFinder(newPage);
+      const oldImages = this.imageFinder(oldPage);
+
+      const imagesToDelete = oldImages.filter(
+        img => newImages.findIndex(newImg => newImg === img) === -1
+      );
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
```

Comment:
Please, check that all old images are deleted properly and if azure returns some errors than return a proper error to the client that updating of business text is failed